Quadriga/src/main/java/edu/asu/spring/quadriga/validator/ProjectURLValidator.java

```diff
@@ -0,0 +1,84 @@
+package edu.asu.spring.quadriga.validator;
+
+import java.util.regex.Pattern;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.validation.Errors;
+import org.springframework.validation.Validator;
+
+import edu.asu.spring.quadriga.accesschecks.IProjectSecurityChecker;
+import edu.asu.spring.quadriga.domain.impl.workbench.Project;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+
+@Service
+public class ProjectURLValidator implements Validator {
+
+    @Autowired
+    IProjectSecurityChecker projectCheckSecurityManager;
+
+    private static final Logger logger = LoggerFactory.getLogger(ProjectURLValidator.class);
+
+    @Override
+    public boolean supports(Class<?> arg0) {
+        return arg0.isAssignableFrom(Project.class);
+    }
+
+    @Override
+    public void validate(Object obj, Errors err) {
+        // validate all the input parameters
+        Project project = (Project) obj;
+
+        String projUnixName = project.getUnixName();
+        String projectId = project.getProjectId();
+
+        if (err.getFieldError("unixName") == null) {
+            // validate the regular expression
+            validateUnixNameExp(projUnixName, err);
+        }
+
+        if (err.getFieldError("unixName") == null) {
+            try {
+                validateUnixName(projUnixName, projectId, err);
+            } catch (QuadrigaStorageException e) {
+                logger.error("Error", e);
+            }
+        }
+    }
+
+    /**
+     * This method validates if the Unix name contains letters other than the
+     * mentioned value in the regular expression
+     * 
+     * @param unixName
+     * @param err
+     * @author kiran batna
+     */
+    public void validateUnixNameExp(String unixName, Errors err) {
```

Comment:
Given that this method and the validateUnixnName method are the same as the ones in AddProjectValidator, it makes sense to put them in their own class und use the class in both both validators. To avoid problem like currently, where you changed one validateUnixName method to not include "!*'" but not the other :)